include/pagmo/archipelago.hpp

```diff
@@ -140,7 +140,7 @@ PAGMO_DLL_PUBLIC std::ostream &operator<<(std::ostream &, migrant_handling);
 class PAGMO_DLL_PUBLIC archipelago
 {
     // Make friends with island.
-    friend class PAGMO_DLL_PUBLIC island;
+    friend class island;
```

Comment:
Could you explain what kind of warnings the ``PAGMO_DLL_PUBLIC`` attribute is giving?

As far as I remember, the original motivation for repeating ``PAGMO_DLL_PUBLIC`` here was that MSVC would complain if a class is not declared everywhere with the same exact set of attributes. But perhaps I am misremembering...

Reply:
its the Wattributes warning
I dont see any warnings in the windows compilation log

Reply:
maybe it was for an older version of msvc ?

Reply:
Possibly... in any case, sorry for taking this long, merging this now.